csharp/extractor/Semmle.Extraction.CSharp/Entities/Expressions/MemberAccess.cs

```diff
@@ -71,7 +71,9 @@ static Expression Create(ExpressionNodeInfo info, ExpressionSyntax expression, S
             if (symbol == null)
             {
                 info.Context.ModelError(info.Node, "Failed to determine symbol for member access");
-                return new MemberAccess(info.SetKind(ExprKind.UNKNOWN), expression, symbol);
+                // Default to property access - this can still give useful results but
+                // the target of the expression should be checked in QL.
+                return new MemberAccess(info.SetKind(ExprKind.PROPERTY_ACCESS), expression, symbol);
```

Comment:
The choice is between wrong some of the time or missing all of the time. So I would argue that the change I made is a step in the right direction that is unlikely to make things worse. I think your changes seem reasonable but I don't think this PR is the place for them, particularly since dataflow libraries and queries would need to be reworked.